extensions/oidc/runtime/src/main/java/io/quarkus/oidc/TenantConfigResolver.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.oidc;
+
+import io.quarkus.oidc.runtime.OidcTenantConfig;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * <p>
+ * A tenant resolver is responsible for resolving the {@link OidcTenantConfig} for tenants, dynamically.
+ * 
+ * <p>
+ * Instead of implementing a {@link TenantResolver} that maps the tenant configuration based on an identifier and its
+ * corresponding entry in the application configuration file, beans implementing this interface can dynamically construct the
+ * tenant configuration without having to define each tenant in the application configuration file.
+ */
+public interface TenantConfigResolver {
```

Comment:
Hi Pedro @pedroigor, this is a step in the right direction IMHO :-). I've been thinking today how to pre-calculate the connections, similarly to the way it is demoed with the Vertx multi-tenancy handler.
So here is a proposal for this interface:
```
public interface TenantConfgResolver extends TenantResolver {
    // Map key is the tenant id 
    Map<String, OidcTenantConfig> getTenantConfigurations();
}
```
Given this interface, the default resolver will precalculate all OAuth2Auth objects and cache them in the recorder, and at runtime, once the tenant id is resolved, it will return TenantConfigContext. Users won't have to implement,  only if they want to.

How does it sound ? Otherwise, in order to avoid recalculating `OAuth2Auth`, we need to have hashcode equals/impementations on the OidcTenantConfig, etc...

Reply:
I'm not sure ... Maybe if we get more feedback and any requirement that justify doing this?

I think the current strategy of caching the tenant context is enough ... However, something is missing here. We need to add some concurrency logic when creating the tenant context for the first time for a given tenant.

Reply:
@pedroigor Hi Pedro, just got back... The proposed modification does not introduce anything new but the option to precalculate the connections. 
If the user sets all the multi-tenant options in `application.properties` then `OidcRecorder` does it and at runtime the connections are already available. 
If the user goes with `TenantConfgResolver` then the connection is established at runtime, it is cached though.
If we can have a case where `TenantConfgResolver` can provided an open set of `OidcTenantConfig` then my proposal won't make sense.
If `TenantConfigResolver` implementation is provided solely because putting the tenant config into `application.properties` is not possible, then there is a performance loss compared to a case where `application.properties` are used.
So I'm not exactly sure what can be improved but it feels like something can be improved :-)

OK, here it is: can we rename `TenantConfgResolver` in your current version to `RuntimeTenantConfgResolver` ? If we do it then we can quickly add later on  `StaticTenantConfgResolver` typed as per my proposal above to address the case where a set of tenant configurations is not open ended (which would be typical enough I guess), we'd just have `OidcRecorder` check an `StaticTenantConfgResolver` bean and deal with it the same way it does with the tenant config in `application.properties`. 

So I'd be happy with just `TenantConfgResolver` to `RuntimeTenantConfgResolver` name change. I can add `StaticTenantConfgResolver` later, or if you don't mind, add it now too. I think we'd cover almost 100% of all the cases here.